app/api/routes/user.py

```diff
@@ -37,3 +47,166 @@ async def create_user(user: UserCreateRequestModel):
+
+    #Authenticated user is trying to update their own information
+    if str(user_id) != str(current_user.id):
+        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail="You can only update your own details.")
+    
+    user = users_db.get(str(user_id))
+    if not user:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="User not found.")
+    
+    #Validate and update the email if provided
+    if update_data.email:
+        if any(existing_user.email == update_data.email for existing_user in users_db.values() if existing_user.id != user_id):
+            raise HTTPException(status_code=status.HTTP_409_CONFLICT, detail="Email already registered.")
+        user.email = update_data.email
+
+    if update_data.username:
+        user.username = update_data.username
+
+    if update_data.password:
+        if not is_valid_password(update_data.password):
+            raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail="Password does not meet the security requirements.")
+        user.hashed_password = get_password_hash(update_data.password)
+
+    user.updated_at = datetime.now(timezone.utc)
+
+    users_db[str(user.id)] = user
+
+    return UserUpdateResponseModel(
+        id=user.id,
+        username=user.username,
+        email=user.email,
+        is_admin=user.is_admin,
+        is_active=user.is_active,
+        created_at=user.created_at,
+        updated_at=user.updated_at
+    )
+
+
+
+#DELETE User
+@router.delete("/users/{user_id}", status_code=status.HTTP_204_NO_CONTENT)
+async def delete_user(user_id: UUID, current_user: User = Depends(get_current_user)):
+    
+    if str(user_id) != str(current_user.id):
+        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail="You can only delete your own account.")
+
+    user = users_db.get(str(user_id))
+    if not user:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="User not found.")
+
+    """
+    This part will be for checking active orders
+    """
+
+    del users_db[str(user_id)]
+    #Return 204 no content
+    return
+
+
+#GET all users 
+@router.get("/users", response_model=list[UserCreateResponseModel], status_code=status.HTTP_200_OK)
+async def get_all_users(current_admin: User = Depends(get_current_admin)):
+
+    try:
+        all_users = list(users_db.values())
+
+        return [
+            UserUpdateResponseModel(
+                id=user.id,
+                username=user.username,
+                email=user.email,
+                is_admin=user.is_admin,
+                is_active=user.is_active,
+                created_at=user.created_at,
+                updated_at=user.updated_at
+            )
+            for user in all_users
+        ]
+
+    except Exception as e:
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"An error occurred while retrieving users: {str(e)}"
```

Comment:
You should never return error details to the user.

Reply:
Yes you are right, i thhink it's work now see it.